external-crates/move/crates/move-compiler/src/shared/mod.rs

```diff
@@ -602,6 +604,20 @@ impl CompilationEnv {
     pub fn primitive_definer(&self, t: N::BuiltinTypeName_) -> Option<&E::ModuleIdent> {
         self.prim_definers.get(&t)
     }
+
+    pub fn has_parsing_error(&self) -> bool {
+        self.parsing_error.is_some()
+    }
+
+    pub fn set_parsing_error(&mut self, diag: Diagnostic) {
```

Comment:
Not really a fan of this. Why not just reprot all of it?

Reply:
I don't like the idea of reporting all other potential problems after the original parsing error. This could potentially be quite confusing if folks started to fix a problem by reading these subsequent diagnostics instead of going straight to the root of the problem. The way it's implemented now, we will get the same parsing error we would be getting if no parsing error recovery was there.

Reply:
If you want to do that, could you manage it in `syntax` itself? That is, wrapping the calls to `add_diag` through the `Context` and checking a local flag? 
It feels weird to pollute the shared env with something specific to parsing